pytorch_lightning/loops/fit_loop.py

```diff
@@ -263,8 +263,9 @@ def teardown(self) -> None:
 
     def on_save_checkpoint(self) -> Dict:
         state_dict = super().on_save_checkpoint()
-        # TODO: update has_completed to its proper value
-        state_dict["dataloader_state_dict"] = self.trainer.train_dataloader.state_dict(has_completed=False)
+        # FIXME(@tchaton) Should pass has_completed=True when iterator is exhausted ?
```

Comment:
I think @awaelchli recently changed this

Reply:
Reverted back.